File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Cards/Status/Slimed.ts
import FileHelper from "../../FileHelper";
import { LocalizationProperty } from "../../Gen/DB/Localization";
import AbstractCard, { CardColor, CardRarity, CardTarget, CardType } from "../AbstractCard";

export class Slimed extends AbstractCard{
    Color: CardColor = CardColor.COLORLESS;
    Desc: string = LocalizationProperty.Read("黏液描述");
    Icon: string = FileHelper.FormatPath("cards/status/slimed.png");
    Name: string = LocalizationProperty.Read("黏液");
    Rarity: CardRarity = CardRarity.COMMON;
    Target: CardTarget = CardTarget.SELF;
    Type: CardType = CardType.STATUS;
    Exhaust = true;
}

File: Unity/Assets/Scripts/ModTheSpire/Card/Status/Slimed.cs

using TCG.Config;

namespace ModTheSpire.Card.Status
{
    public class Slimed : AbstractCard
    {
        public override string Desc { get; } = LocalizationProperty.Read("黏液描述");
        public override string Icon { get; } = "cards/status/slimed";
        public override CardType Type { get; } = CardType.STATUS;
        public override CardRarity Rarity { get; }= CardRarity.COMMON;
        public override CardTarget Target { get; } = CardTarget.SELF;
        public override CardColor Color { get; } = CardColor.COLORLESS;

        public Slimed()
        {
            Exhaust = true;
        }
    }
}